### PR TITLE
Remove heic support from media exporter

### LINKED ISSUE #21617: Upload pictures from Iphone not showing in website or in gallery
### Expected behavior
convert heic photo to jpg and upload. In the past everyting was ok. 
Captură de ecran din 2023-09-21 la 14 00 22
Captură de ecran din 2023-09-21 la 14 00 15

### Actual behavior
not showing pictures  in website or Media Gallery

### PR DESCRIPTION
**Pull request by kean**
Fixes #21617

## To test:

- Create a self-hosted site (or use an existing one)
- Open the app and go to Media
- Upload a photo taken on an iPhone/iPad (has to be in `.heic` format)
- Verify that the photo got uploaded
- Open the WP admin page and verify that the photo is [displayed](https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/0d25648b-c154-4a65-99e7-8cd026ded430) and has `.jpeg` format

The app uses the same logic for uploads in all areas of the app, but it may also be worth checking other screens, at least the editor.

## Regression Notes
1. Potential unintended areas of impact: Media uploads
2. What I did to test those areas of impact (or what existing automated tests I relied on): manual
3. What automated tests I added (or what prevented me from doing so): n/a

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by kean** [comment]
Hey, @oguzkocer, just a heads-up: we are targeting 23.3 for this bug fix. Thanks.

**Comment by kean** [comment]
Hey, @mokagio, just a heads-up: we are targeting 23.3 for this bug fix. Thanks.

**Review by momo-ozawa — APPROVED** [review]
LGTM! Verified .heic photos got uploaded via media / editor screens. wp-admin shows them as .jpeg images.

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -11,6 +11,7 @@
 * [**] [internal] Refactor fetching one single media item. [#21567]
 * [**] [internal] Refactored how the total number of WordPress Media is fetched. [#21568]
 * [**] [internal] Refactor fetching metadata of VideoPress videos. [#21569]
+* [*] Fix support for `.heic` uploads to self-hosted sites [#21622]
```
Comment (momo-ozawa) [inline RELEASE-NOTES.txt]: I think there's a separate process to update release notes after code freeze

Comment (momo-ozawa) [reply RELEASE-NOTES.txt]: ref: p77Llu-bGf-p2

Comment (kean) [reply RELEASE-NOTES.txt]: Removed for now thanks

**Pull request merged**